### PR TITLE
feat: Name History

### LINKED ISSUE #73: Name history
The name history of a long name (13+ characters) that doesn’t need an auction could look as follows:

image

The name history of a short name (<13 characters) that needs an auction could look as follows:

image

Other name activities:

image

Open questions:
* How to find out if NameClaimTx is: an “auction initializer”, “auction bid” or just “name activation” in case there is no auction?
* How to find out that the auction took place and has ended + easily define who won it?
* How to find out if the name is expired?

Tracking: https://github.com/aeternity/ae_mdw/issues/1202

Sample endpoint: https://mainnet.aeternity.io/mdw/v2/names/superhero.chain/history

### PR DESCRIPTION
**Pull request by janmichek**
## Description
resolves #73 

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] add Name History

### CODE REVIEW

**Comment by janmichek** [comment]
If you agree, we can do the review here 

Later solve the MDW blocked parts separately https://github.com/aeternity/aescan/issues/525 and

**Comment by janmichek** [comment]
@michele-franchi fixed
image

**Review by Liubov-crypto — APPROVED** [review]
LGTM. 

@janmichek I have a question: this acronym "TTL" visually get looked to me as merged together. Is this acceptable? Will we decrypt this acronym for the end user?

ttl

File: src/components/NameHistoryDataCellNameClaimEvent.vue [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      {{ formatAePrice(formatAettosToAe(payload.tx.nameFee), null) }}
+    </app-chip>
+    <template #tooltip>
+      Fee
+    </template>
+  </app-tooltip>
+</template>
+
+<script setup>
+import ValueHashEllipsed from '~/components/ValueHashEllipsed'
```
Comment (michele-franchi) [inline src/components/NameHistoryDataCellNameClaimEvent.vue:17]:
```suggestion
import ValueHashEllipsed from '@/components/ValueHashEllipsed'
```

File: src/components/NameHistoryDataCellNameRevokeTxEvent.vue [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<template>
+  <value-hash-ellipsed
+    :hash="payload.tx.accountId"
+    :link-to="`/accounts/${payload.tx.accountId}`"/>
+</template>
+
+<script setup>
+import ValueHashEllipsed from '~/components/ValueHashEllipsed'
```
Comment (michele-franchi) [inline src/components/NameHistoryDataCellNameRevokeTxEvent.vue:8]:
```suggestion
import ValueHashEllipsed from '@/components/ValueHashEllipsed'
```

File: src/components/NameHistoryDataCellNameTransferTxEvent.vue [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  <transaction-arrow-right-icon/>
+
+  <value-hash-ellipsed
+    :hash="payload.tx.recipientId"
+    :link-to="`/accounts/${payload.tx.recipientId}`"/>
+</template>
+
+<script setup>
+import ValueHashEllipsed from '~/components/ValueHashEllipsed'
```
Comment (michele-franchi) [inline src/components/NameHistoryDataCellNameTransferTxEvent.vue:14]:
```suggestion
import ValueHashEllipsed from '@/components/ValueHashEllipsed'
```

File: src/components/NameHistoryDataCellNameUpdateTxEvent.vue [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    </app-chip>
+    <template #tooltip>
+      TTL
+    </template>
+  </app-tooltip>
+</template>
+
+<script setup>
+import AppChip from '~/components/AppChip'
+import ValueHashEllipsed from '~/components/ValueHashEllipsed'
```
Comment (michele-franchi) [inline src/components/NameHistoryDataCellNameUpdateTxEvent.vue:18]:
```suggestion
import AppChip from '@/components/AppChip'
import ValueHashEllipsed from '@/components/ValueHashEllipsed'
```

File: src/components/NameHistoryTable.vue [hunk 5]
```diff
@@ -28,26 +31,34 @@
...
       </tr>
     </tbody>
   </table>
 </template>
 
 <script setup>
-import ValueHashEllipsed from '@/components/ValueHashEllipsed'
 import { namesHints } from '@/utils/hints/namesHints'
 import HintTooltip from '@/components/HintTooltip'
 import BlockTimeCell from '~/components/BlockTimeCell'
```
Comment (michele-franchi) [inline src/components/NameHistoryTable.vue:64]:
```suggestion
import BlockTimeCell from '@/components/BlockTimeCell'
```

**Review by michele-franchi — APPROVED** [review]
Looks good, I added suggestions for imports.

I also notice the pipeline is failing due to a different version of Node.js, could you please check it with Petar?

File: src/components/NameHistoryTable.vue (hunk 5 above)
Comment (janmichek) [reply src/components/NameHistoryTable.vue:64]: Ooops, forgot this. All imports fixed

**Comment by janmichek** [comment]
> Looks good, I added suggestions for imports.
> 
> I also notice the pipeline is failing due to a different version of Node.js, could you please check it with Petar?

WIll check it

**Comment by janmichek** [comment]
> > Looks good, I added suggestions for imports.
> > I also notice the pipeline is failing due to a different version of Node.js, could you please check it with Petar?
> 
> WIll check it

All fixed

**Pull request merged**